Reconnecting to open app

WojtekKowaluk opened the pull request:
This is based on PR #20. I have rebased on up-to-date master also fixed problem which caused freeze between test suites in acceptance tests.

Reason for freeze was excessive waiting in `get_keyword_names`. Total waiting time for operation to timeout was about 30 minutes instead of expected 60 seconds. This change may also fix know issue when `Start Application` does not timeout in some circumstances when it has failed to connect to agent.

After I fixed that I discovered that in suite where RemoteSwingLibrary is imported with different settings state of library was not "clean" and this is why timeout happened in first place.

Solution is to handle two kinds of reimports differently: reimport which refresh keyword list after connecting to agent and reimport with different settings. Because RLS uses some static fields and a global variable (to keep state in first type of reimport), we need to reset them on second type of reimport.

jussimalinen commented:
Good work. To comment a minor detail, there isnt a consistency in apport and aphost names. Sometimes they are spelled apphost, sometimes aphost for example. Consistent naming would be nice.

The tests in existing_app_1 and existing_app_2 are of course run with the same robot instance. Did you verify somehow (once manually is fine I guess) that this works between different robot runs? I suppose that would be one of the main usages? (Assuming that the first robot wont kill the process for the started app)

WojtekKowaluk commented:
Good find, I this inconsistency is just typo in docstring, fixed.

This is exactly why i have split original test suite from Jakub into two. This use case can be tested manually by calling robot with first suite, then calling robot again with second one. Of course this property is not tested when we call `mvn clean verify`, do you think something should be done about it?

stasiek commented:
@jussimalinen Do You think we can go on and merge it?
